src/cnc/flavors/aws/lambda-lite/1/provision/partials/collection/vpc.tf.j2

```diff
@@ -0,0 +1,294 @@
+
+  domain = "vpc"
+
+  tags = merge(
+    {
+      "Name" = format(
+        "${local.{{ environment.name }}_{{environment.collection.instance_name}}_vpc_infrastructure_suffix}-%s",
+        element(var.{{ environment.name }}_{{environment.collection.instance_name}}_azs, var.{{ environment.name }}_{{environment.collection.instance_name}}_single_nat_gateway ? 0 : count.index),
+      )
+    },
+    var.{{ environment.name }}_{{environment.collection.instance_name}}_nat_eip_tags,
+  )
+
+  depends_on = [aws_internet_gateway.{{ environment.name }}_{{environment.collection.instance_name}}]
+}
+
+resource "aws_nat_gateway" "{{ environment.name }}_{{environment.collection.instance_name}}" {
+  count = var.{{ environment.name }}_{{environment.collection.instance_name}}_enable_nat_gateway ? local.{{ environment.name }}_{{environment.collection.instance_name}}_nat_gateway_count : 0
+
+  allocation_id = element(
+    local.{{ environment.name }}_{{environment.collection.instance_name}}_nat_gateway_ips,
+    var.{{ environment.name }}_{{environment.collection.instance_name}}_single_nat_gateway ? 0 : count.index,
+  )
+  subnet_id = element(
+    aws_subnet.{{ environment.name }}_{{environment.collection.instance_name}}_public[*].id,
+    var.{{ environment.name }}_{{environment.collection.instance_name}}_single_nat_gateway ? 0 : count.index,
+  )
+
+  tags = merge(
+    {
+      "Name" = format(
+        "${local.{{ environment.name }}_{{environment.collection.instance_name}}_vpc_infrastructure_suffix}-%s",
+        element(var.{{ environment.name }}_{{environment.collection.instance_name}}_azs, var.{{ environment.name }}_{{environment.collection.instance_name}}_single_nat_gateway ? 0 : count.index),
+      )
+    },
+    var.{{ environment.name }}_{{environment.collection.instance_name}}_nat_gateway_tags,
+  )
+
+  depends_on = [aws_internet_gateway.{{ environment.name }}_{{environment.collection.instance_name}}]
+}
+
+resource "aws_route" "{{ environment.name }}_{{environment.collection.instance_name}}_private_nat_gateway" {
+  count = var.{{ environment.name }}_{{environment.collection.instance_name}}_enable_nat_gateway ? local.{{ environment.name }}_{{environment.collection.instance_name}}_nat_gateway_count : 0
+
+  route_table_id         = element(aws_route_table.{{ environment.name }}_{{environment.collection.instance_name}}_private[*].id, count.index)
+  destination_cidr_block = var.{{ environment.name }}_{{environment.collection.instance_name}}_nat_gateway_destination_cidr_block
+  nat_gateway_id         = element(aws_nat_gateway.{{ environment.name }}_{{environment.collection.instance_name}}[*].id, count.index)
+
+  timeouts {
+    create = "5m"
+  }
+}
+
+resource "aws_vpc_endpoint" "{{ environment.name }}_{{environment.collection.instance_name}}_dynamodb" {
```

Comment:
Do we only need this if we have dynamodb instances?

Reply:
Ajustments Commit: https://github.com/tmaior/cnc-fork/commit/9ed6e46909e2e43c43eee39bb138ed6d2df70b18